flink-libraries/flink-table/src/main/java/org/apache/flink/table/api/catalog/ReadableWritableCatalog.java

```diff
@@ -0,0 +1,175 @@
+		throws DatabaseNotExistException;
+
+	// ------ tables and views ------
+
+	/**
+	 * Deletes a table or view.
+	 *
+	 * @param objectName         Path of the table or view to delete.
+	 * @param ignoreIfNotExists Flag to specify behavior if the table or view does not exist:
+	 *                          if set to false, throw an exception,
+	 *                          if set to true, nothing happens.
+	 * @throws TableNotExistException    thrown if the table or view does not exist
+	 */
+	void dropTable(ObjectPath objectName, boolean ignoreIfNotExists) throws TableNotExistException;
+
+
+	// ------ tables ------
+
+	/**
+	 * Create a new table in the catalog.
+	 * Note that TableStats in the CatalogTable will not be used for creation.
+	 * Use {@link #alterTableStats(ObjectPath, TableStats, boolean)} to alter table stats.
+	 *
+	 * @param tableName      Path of the table to create.
+	 * @param table          The table to create.
+	 * @param ignoreIfExists Flag to specify behavior if a table with the given name already exists:
+	 *                       if set to false, it throws a TableAlreadyExistException,
+	 *                       if set to true, nothing happens.
+	 * @throws TableAlreadyExistException thrown if table already exists and ignoreIfExists is false
+	 * @throws DatabaseNotExistException thrown if the database that this table belongs to doesn't exist
+	 */
+	void createTable(ObjectPath tableName, CatalogTable table, boolean ignoreIfExists)
+		throws TableAlreadyExistException, DatabaseNotExistException;
+
+	/**
+	 * Modifies an existing table.
+	 * Note that TableStats in the CatalogTable will not be used for alteration.
+	 * Use {@link #alterTableStats(ObjectPath, TableStats, boolean)} to alter table stats.
+	 *
+	 * @param tableName         Path of the table to modify.
+	 * @param newTable             The new table which replaces the existing table.
+	 * @param ignoreIfNotExists Flag to specify behavior if the table does not exist:
+	 *                          if set to false, throw an exception,
+	 *                          if set to true, nothing happens.
+	 * @throws TableNotExistException   thrown if the table does not exist
+	 */
+	void alterTable(ObjectPath tableName, CatalogTable newTable, boolean ignoreIfNotExists)
+		throws TableNotExistException;
+
+	/**
+	 * Renames an existing table.
+	 *
+	 * @param tableName        Path of the table to modify.
+	 * @param newTableName     New name of the table.
+	 * @param ignoreIfNotExists Flag to specify behavior if the table does not exist:
+	 *                          if set to false, throw an exception,
+	 *                          if set to true, nothing happens.
+	 * @throws TableNotExistException thrown if the table does not exist
+	 * @throws DatabaseNotExistException thrown if the database that this table belongs to doesn't exist
+	 */
+	void renameTable(ObjectPath tableName, String newTableName, boolean ignoreIfNotExists)
```

Comment:
does it need to throw a 'TableAlreadyExistsException' when the 'newTableName' is already taken?

Reply:
I think an explanation similar to that for "DatabaseAlreadyExistsException" can be given here too.